### ISSUE TITLE
Failure creating loom user on server

### ISSUE THREAD

**Issue by isaacliao** [StanfordBioinformatics/loom#237]
```
TASK [Create SSH key if it doesn't exist] **************************************
task path: /opt/loom/loomengine/playbooks/tasks/gcloud_configure_instance_ssh.yml:1
Using module file /opt/loom/local/lib/python2.7/site-packages/ansible/modules/core/system/user.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1478723783.15-88647264006149 `" && echo ansible-tmp-1478723783.15-88647264006149="` echo $HOME/.ansible/tmp/ansible-tmp-1478723783.15-88647264006149 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpM6R6nu TO /root/.ansible/tmp/ansible-tmp-1478723783.15-88647264006149/user.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1478723783.15-88647264006149/ /root/.ansible/tmp/ansible-tmp-1478723783.15-88647264006149/user.py && sleep 0'
Using module file /opt/loom/local/lib/python2.7/site-packages/ansible/modules/core/system/user.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1478723783.39-209750736039826 `" && echo ansible-tmp-1478723783.39-209750736039826="` echo $HOME/.ansible/tmp/ansible-tmp-1478723783.39-209750736039826 `" ) && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/opt/loom/bin/python /root/.ansible/tmp/ansible-tmp-1478723783.15-88647264006149/user.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1478723783.15-88647264006149/" > /dev/null 2>&1 && sleep 0'
<127.0.0.1> PUT /tmp/tmpscqJlP TO /root/.ansible/tmp/ansible-tmp-1478723783.39-209750736039826/user.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1478723783.39-209750736039826/ /root/.ansible/tmp/ansible-tmp-1478723783.39-209750736039826/user.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/opt/loom/bin/python /root/.ansible/tmp/ansible-tmp-1478723783.39-209750736039826/user.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1478723783.39-209750736039826/" > /dev/null 2>&1 && sleep 0'
fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_args": {
            "append": false,
            "comment": null,
            "createhome": true,
            "expires": null,
            "force": false,
            "generate_ssh_key": true,
            "group": null,
            "groups": null,
            "home": null,
            "login_class": null,
            "move_home": false,
            "name": "loom",
            "non_unique": false,
            "password": null,
            "remove": false,
            "seuser": null,
            "shell": null,
            "skeleton": null,
            "ssh_key_bits": 2048,
            "ssh_key_comment": "ansible-generated on isaac-roles",
            "ssh_key_file": "/root/.ssh/google_compute_engine",
            "ssh_key_passphrase": null,
            "ssh_key_type": "rsa",
            "state": "present",
            "system": false,
            "uid": null,
            "update_password": "always"
        },
        "module_name": "user"
    },
    "msg": "useradd: user 'loom' already exists\n",
    "name": "loom",
    "rc": 9
}
        to retry, use: --limit @/opt/loom/loomengine/playbooks/gcloud_create_worker.retry

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=4    unreachable=0    failed=1
```
Using this settings file:
```
[gcloud]
DOCKER_TAG = customserviceaccounts
WORKER_USES_SERVER_INTERNAL_IP = True
CLIENT_USES_SERVER_INTERNAL_IP = True

SERVER_NETWORK = development
SERVER_CUSTOM_SUBNET = computes-uscentral
SERVER_LOCATION = us-central1-a
SERVER_TAGS = computes-uscentral,https-server

WORKER_NETWORK = development
WORKER_CUSTOM_SUBNET = computes-uscentral
WORKER_LOCATION = us-central1-a
WORKER_TAGS = computes-uscentral

SERVICE_ACCOUNT_ROLES = ["roles/compute.instanceAdmin", "roles/compute.storageAdmin", "roles/logging.logWriter", "roles/storage.objectCreator", "roles/storage.objectViewer", "roles/logging.viewer", "roles/iam.serviceAccountActor"]
```

**Comment by nhammond** [comment]
This step has been set to ignore errors because of this and a similar error where two steps race to generate an ssh key. 

A better solution would be to move the "Create SSH key" from gcloud_configure_instance_ssh.yml (where it is executed multiple times per machine) to gcloud_create_instance.yml where it will run only once.

**Comment by isaacliao** [comment]
Hmm, from a quick look I don't see why this is executed multiple times per machine. It's called like this:
```
- name: Create a new instance for the Loom server.
  hosts: localhost
  connection: local
  vars_files:
  - vars/common.yml
  - vars/gcloud.yml
  tasks:
  - include: tasks/gcloud_create_instance.yml
  - include: tasks/gcloud_configure_instance_ssh.yml
```
So, if it solves the problem, we could just move the contents of gcloud_configure_instance_ssh.yml to the end of gcloud_create_instance.yml.

**Comment by nhammond** [comment]
You're right, that wasn't very clear. The other important snippet is worker creation:

```
gcloud_run_task.yml:
---
- name: Create a new instance for the Loom worker.
  hosts: localhost
  connection: local
  gather_facts: no
  vars_files:
  - vars/common.yml
  - vars/gcloud_worker.yml
  tasks:
  - include: tasks/gcloud_create_instance.yml
  - include: tasks/gcloud_configure_instance_ssh.yml
...
```

All the commands in "gcloud_create_instance" and "gcloud_configure_instance_ssh" are run on localhost, when bringing up a remote machine, either server or worker.

For client bringing up server, that executes once. For server bringing up workers, it runs many times, and they all share the same server ssh key. Creating that key is where we get errors from race conditions.

We want the to create an ssh key locally on the client once if needed, and add it to the server metadata once after the server comes up. Also once after the server is created, we want to create a key remotely on the server. When each worker is coming up, we need to add the existing key to the instance metadata but not run ssh-keygen again. 

Something like this:
```
gcloud_start_server.yml:
  name: Create a new instance for the Loom server.
  hosts: localhost
    tasks:
  # New: MYSSHKEYGENTASK - create an ssh key if it doesn't exist
  - include: tasks/gcloud_create_instance.yml
  - include: tasks/gcloud_configure_instance_ssh.yml # REMOVE KEYGEN--JUST ADD TO METADATA

# Then run MYSSHKEYGENTASK on "hosts: newinstances"
```
Then for the workers:
```
- name: Create a new instance for the Loom worker.
  hosts: localhost
  connection: local
  gather_facts: no
  vars_files:
  - vars/common.yml
  - vars/gcloud_worker.yml
  tasks:
  - include: tasks/gcloud_create_instance.yml
  - include: tasks/gcloud_configure_instance_ssh.yml # SAME AS BEFORE, BUT WITHOUT KEYGEN
```

**Comment by isaacliao** [comment]
Oh okay, that makes sense. Thanks for the explanation. I'll give it a shot since I'm working on worker playbooks now anyway.

**Comment by isaacliao** [comment]
Implemented in cb430044d6f9d2655082f744673bff444e5c1e3b.
